plasmapy/formulary/parameters.py

```diff
@@ -663,23 +689,27 @@ def thermal_speed(
     >>> thermal_speed(1e6*u.K, "e-", method="mean_magnitude")
     <Quantity 621251... m / s>
     """
-    m = mass if np.isfinite(mass) else particles.particle_mass(particle)
+    if mass is None:
+        mass = particles.particle_mass(particle)
 
     # different methods, as per https://en.wikipedia.org/wiki/Thermal_velocity
     try:
-        coef = _coefficients[ndim]
+        coeff = thermal_speed_coefficients[ndim]
     except KeyError:
         raise ValueError("{ndim} is not a supported value for ndim in thermal_speed")
     try:
-        coef = coef[method]
+        coeff = coeff[method]
     except KeyError:
         raise ValueError("Method {method} not supported in thermal_speed")
 
-    return np.sqrt(coef * k_B * T / m)
+    speed = thermal_speed_lite(T=T.value, mass=mass.value, coeff=coeff)
+    return speed * u.m / u.s
 
 
 vth_ = thermal_speed
-""" Alias to :func:`thermal_speed`. """
+""" Alias to :func:`~plasmapy.formulary.parameters.thermal_speed`. """
```

Comment:
Works nicely with #1099, too! :smile:

Reply:
Yeah, I ran across this same thing when working on this PR.  I actually thought I already made this change.  I think we need to get in the practice of using the explicit link since `` `thermal_speed` `` would just look for the variable in the current `global()` scope.